modules/sdk-coin-ton/test/unit/singleNominatorWithdrawBuilder.ts

```diff
@@ -0,0 +1,216 @@
+import should from 'should';
+import { TransactionType } from '@bitgo/sdk-core';
+import { TransactionBuilderFactory } from '../../src/lib/transactionBuilderFactory';
+import { coins } from '@bitgo/statics';
+import * as testData from '../resources/ton';
+import { KeyPair } from '../../src/lib/keyPair';
+import * as utils from '../../src/lib/utils';
+import TonWeb from 'tonweb';
+
+describe('Ton Single Nominator Withdraw Builder', () => {
+  const factory = new TransactionBuilderFactory(coins.get('tton'));
+  it('should build a unsigned withdraw tx', async function () {
+    const txId = 'wxaFRG6b7Fhw91Al3JY-zbvl9Mjr-P0NuLvoGYyS5FI='.replace(/\//g, '_').replace(/\+/g, '-');
+    const txBuilder = factory.getSingleNominatorWithdrawBuilder();
+    txBuilder.sender(testData.sender.address);
+    txBuilder.sequenceNumber(0);
+    txBuilder.publicKey(testData.sender.publicKey);
+    txBuilder.expireTime(1234567890);
+    txBuilder.send(testData.recipients[0]);
+    txBuilder.setWithdrawMessage('1');
```

Comment:
where to set withdraw amount

Reply:
i split the message creation into `setWithdrawAmount` and `setWithdrawMessage` functions